ci/upload_audit_events_s3.py

```diff
@@ -0,0 +1,179 @@
+        self.token = self.get_client_credentials_token()
+
+    def get_client_credentials_token(self):
+        with requests.Session() as s:
+            response = s.post(
+                urljoin(self.UAA_API_URL, "oauth/token"),
+                data={
+                    "grant_type": "client_credentials",
+                    "client_id": self.UAA_CLIENT_ID,
+                    "client_secret": self.UAA_CLIENT_SECRET,
+                    "response_type": "token",
+                },
+                auth=requests.auth.HTTPBasicAuth(
+                    self.UAA_CLIENT_ID, self.UAA_CLIENT_SECRET
+                ),
+                timeout=30,
+            )
+            response.raise_for_status()
+            return response.json()["access_token"]
+
+    @functools.cache
+    def get_cf_entity_name(self, entity_path, entity_guid):
+        """
+        Retrieves the name of a CF entity from a GUID.
+        """
+        if not entity_guid:
+            return
+
+        with requests.Session() as s:
+            s.headers["Authorization"] = f"Bearer {self.token}"
+            url = urljoin(self.CF_API_URL, f"v3/{entity_path}/{entity_guid}")
+            response = s.get(url)
+
+            if response.status_code == 404:
+                return
+
+            data = response.json()
+            return data["name"]
+
+    def get_audit_logs(self, start, end):
+        audit_logs = []
+
+        with requests.Session() as s:
+            s.headers["Authorization"] = f"Bearer {self.token}"
+            params = {
+                "created_ats[gt]": str(start),
+                "created_ats[lt]": str(end),
+                "order_by": "created_at",
+            }
+            url = urljoin(self.CF_API_URL, "/v3/audit_events")
+
+            first_response = s.get(url, params=params)
+            data = first_response.json()
+            audit_logs.extend(data["resources"])
+
+            while data["pagination"]["next"] is not None:
+                data = s.get(data["pagination"]["next"]["href"]).json()
+                audit_logs.extend(data["resources"])
+
+        return audit_logs
+
+    def transform_audit_event(self, audit_event):
+        # remove "links" property from event
+        transformed_event = {k: v for k, v in audit_event.items() if k not in ["links"]}
+
+        organization = audit_event.get("organization", {})
+        if organization:
+            organization_name = self.get_cf_entity_name(
+                "organizations",
+                organization.get("guid", None),
+            )
+            if organization_name:
+                transformed_event["organization_name"] = organization_name
+
+        space = audit_event.get("space", {})
+        if space:
+            space_name = self.get_cf_entity_name(
+                "spaces",
+                space.get("guid", None),
+            )
+            if space_name:
+                transformed_event["space_name"] = space_name
```

Comment:
Same thing, but here I'll show the maybe-too-much version:
```suggestion    
        if space_guid := audit_event.get("space", {}).get("guid"):
            if space_name := self.get_cf_entity_name("spaces", space_guid):
                transformed_event["space_name"] = space_name
```